python/lsst/ts/watcher/alarm.py

```diff
@@ -0,0 +1,205 @@
+# This file is part of ts_watcher.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["Alarm"]
+
+import time
+
+from lsst.ts.idl.enums.Watcher import AlarmSeverity
+from lsst.ts import salobj
+
+
+class Alarm:
+    """A Watcher alarm.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of alarm. This must be unique among all alarms
+        and should be of the form system_[subsystem_...]_name
+        so that groups of related alarms can be acknowledged.
+        Use underscores instead of dots to make the regular expressions
+        or glob expressions easier to write.
+    callback : ``callable``
+        Function or coroutine to call whenever the alarm changes state.
+    """
+    def __init__(self, name, callback):
+        self.name = name
+        self.callback = callback
+
+        self.severity = AlarmSeverity.NONE
+        self.max_severity = AlarmSeverity.NONE
+        self.reason = ""
+        self.acknowledged = False
+        self.acknowledged_by = ""
+        self.enabled = True
+
+        self.timestamp_severity_oldest = 0
+        self.timestamp_severity_newest = 0
+        self.timestamp_max_severity = 0
+        self.timestamp_acknowledged = 0
+
+    def __eq__(self, other):
+        for field in ("name",
+                      "callback",
+                      "severity",
+                      "max_severity",
+                      "reason",
+                      "acknowledged",
+                      "acknowledged_by",
+                      "enabled",
+                      "timestamp_severity_oldest",
+                      "timestamp_severity_newest",
+                      "timestamp_max_severity",
+                      "timestamp_acknowledged"):
+            if getattr(self, field) != getattr(other, field):
+                return False
+        return True
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def set_severity(self, severity, reason):
```

Comment:
So assuming that if an alarm goes from severity.none to another severity - it has become "active".  LTS-897-REQ-0001 then requires a SAL event to be published if that is the case - maybe this happens elsewhere?

Reply:
Yes. Again the alarm state will change and the Alarm event will be published, but not by the `Alarm` class.